perfkitbenchmarker/providers/azure/azure_network.py

```
@@ -21,264 +21,284 @@
+    # group itself needs to have a location. Therefore,
+    # FLAGS.zones[0].
+    _, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'group', 'create',
+         self.name, self.location])
 
-  def _Delete(self):
-    """Deletes the affinity group."""
-    delete_cmd = [AZURE_PATH,
-                  'account',
-                  'affinity-group',
-                  'delete',
-                  '--quiet',
-                  self.name]
-    vm_util.IssueCommand(delete_cmd)
+    if retcode:
+      raise errors.RetryableCreationError(
+          'Error creating Azure resource group')
 
   def _Exists(self):
-    """Returns true if the affinity group exists."""
-    show_cmd = [AZURE_PATH,
-                'account',
-                'affinity-group',
-                'show',
-                '--json',
-                self.name]
-    stdout, _, _ = vm_util.IssueCommand(show_cmd, suppress_warning=True)
-    try:
-      json.loads(stdout)
-    except ValueError:
-      return False
-    return True
+    _, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'resource', 'list', self.name])
+
+    return retcode == 0
+
+  def _Delete(self):
+    vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'group', 'delete',
+         '--quiet',
+         self.name])
 
 
 class AzureStorageAccount(resource.BaseResource):
   """Object representing an Azure Storage Account."""
 
-  def __init__(self, name, storage_type, affinity_group_name):
+  def __init__(self, storage_type, location, name,
+               kind=None, access_tier=None):
     super(AzureStorageAccount, self).__init__()
-    self.name = name
     self.storage_type = storage_type
-    self.affinity_group_name = affinity_group_name
+    self.name = name
+    self.resource_group = GetResourceGroup()
+    self.location = location
+    self.kind = kind or 'Storage'
+    self.access_tier = (access_tier or 'Hot') if kind == 'BlobStorage' else None
```

Hrm, this looks somewhat complicated.  If someone only passes in an `access_tier` they have to pass in `kind='BlobStorage'` in order to get `self.access_tier` to be set to a non-null?  I suppose that's okay but maybe we should throw an exception if we're setting something to None when they might think it will be set?